src/simulation/m_lag_bubbles.fpp

```diff
@@ -0,0 +1,1746 @@
+        @:ALLOCATE(lag_bub_gas_betaT(1:lag_nBubs_glb))
+        @:ALLOCATE(lag_bub_gas_betaC(1:lag_nBubs_glb))
+        @:ALLOCATE(lag_bub_dphidt(1:lag_nBubs_glb))
+        @:ALLOCATE(lag_bub_gas_p(1:lag_nBubs_glb, 1:2))
+        @:ALLOCATE(lag_bub_gas_mv(1:lag_nBubs_glb, 1:2))
+        @:ALLOCATE(lag_bub_interface_rad(1:lag_nBubs_glb, 1:2))
+        @:ALLOCATE(lag_bub_interface_vel(1:lag_nBubs_glb, 1:2))
+        @:ALLOCATE(lag_bub_motion_pos(1:lag_nBubs_glb, 1:3, 1:2))
+        @:ALLOCATE(lag_bub_motion_posPrev(1:lag_nBubs_glb, 1:3, 1:2))
+        @:ALLOCATE(lag_bub_motion_vel(1:lag_nBubs_glb, 1:3, 1:2))
+        @:ALLOCATE(lag_bub_motion_s(1:lag_nBubs_glb, 1:3, 1:2))
+        @:ALLOCATE(lag_bub_interface_draddt(1:lag_nBubs_glb, 1:6))
+        @:ALLOCATE(lag_bub_interface_dveldt(1:lag_nBubs_glb, 1:6))
+        @:ALLOCATE(lag_bub_gas_dpdt(1:lag_nBubs_glb, 1:6))
+        @:ALLOCATE(lag_bub_gas_dmvdt(1:lag_nBubs_glb, 1:6))
+        @:ALLOCATE(lag_bub_motion_dposdt(1:lag_nBubs_glb, 1:3, 1:6))
+        @:ALLOCATE(lag_bub_motion_dveldt(1:lag_nBubs_glb, 1:3, 1:6))
+
+        !< Allocate space for the RKCK 4th/5th time stepper coefficients
+        @:ALLOCATE(lag_RKcoef(1:7, 1:6))
+        do i = 1, 6
+            lag_RKcoef(1, i) = RKcoef1(i)
+            lag_RKcoef(2, i) = RKcoef2(i)
+            lag_RKcoef(3, i) = RKcoef3(i)
+            lag_RKcoef(4, i) = RKcoef4(i)
+            lag_RKcoef(5, i) = RKcoef5(i)
+            lag_RKcoef(6, i) = RKcoef6(i)
+            lag_RKcoef(7, i) = RKcoefE(i)
+        end do
+
+        ! Reading bubbles
+        call s_read_input_lag_bubbles(q_cons_vf)
+
+    end subroutine s_initialize_lag_bubbles
+
+    !> The purpose of this procedure is to read the input file with the bubbles' information
+        !! @param q_cons_vf Conservative variables
+    subroutine s_read_input_lag_bubbles(q_cons_vf)
+
+        type(scalar_field), dimension(sys_size), intent(inout) :: q_cons_vf
+
+        real(kind(0.d0)), dimension(8) :: inputparticle
+        real(kind(0.d0)) :: qtime
+        integer :: id, nparticles, save_count
+        integer :: i, j, k, l
+        logical :: file_exist, indomain
+
+        character(LEN=path_len + 2*name_len) :: path_D_dir !<
+
+        dt_max = dt !Initial and largest dt
+
+        ! Initialize number of particles
+        nparticles = 0
+        id = 0
+
+        ! Read the input lag_bubble file or restart point
+        if (cfl_dt) then
+            save_count = n_start
+            qtime = n_start*t_save
+        else
+            save_count = t_step_start
+            qtime = t_step_start*dt
+        end if
+
+        if (save_count == 0) then
+            if (proc_rank == 0) print *, 'Reading lag_bubbles input file at save_count: ', save_count
+            inquire (file='input/lag_bubbles.dat', exist=file_exist)
+            if (file_exist) then
+                open (unit=85, file='input/lag_bubbles.dat', form='formatted')
+101             read (85, *, end=102) (inputparticle(i), i=1, 8)
```

Comment:
identifier numbers here? i'd like to to stay away with this since we use it no where else in the code and I'm not sure what it might break (now or in the future). can we do without these?

Reply:
I have removed all goto and continue statements. I need a small clarification for this specific comment, when you say identifier numbers, you mean the '101' and '102' numbers right? or do you refer to the file unit identifier number (85)?

Reply:
Right, I mean `101/102` here. Please just follow the usual MFC syntax here.

Reply:
fixed